### PR TITLE
Permissions Management documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Permissions management documentation.

## Type of documentation

- [x] New documentation
- [x] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Permissions Management](https://www.notion.so/rudderstacks/Permissions-Management-a77b40e21a70492394d080024797e1b5)

### CODE REVIEW

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+| <Link to="/get-started/quickstart/">Connections</Link>     | Yes  | No  | No     | No     |
+| <Link to="/dashboard-guides/live-events/">Live Events</Link> | Yes | - | - | - |
+| <Link to="/features/transformations/">Transformations</Link> | Yes  | No  | No     | No     |
+| <Link to="/dashboard-guides/audit-logs/">Audit Logs</Link> | No | - | - | - |
+| <Link to="/features/data-governance/tracking-plans/">Tracking Plans</Link> | Yes  | No  | No     | No     |
+| <Link to="/sources/reverse-etl/features/models/">Models</Link> | Yes | No  | No     | No     |
+
+Some things to note regarding the read-only user permissions:
+- Read-only users can view the settings of all the destinations. However, secrets like access keys are hidden from them.
+- They can run tests on existing transformations but they cannot add, modify, or delete transformations.
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:77]: Do we need this in the docs? I don't see why someone would like to run tests on an existing transformation

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+For detailed role-specific permissions for various RudderStack features and settings, refer to the <Link to="/dashboard-guides/permissions-management/permissions-matrix/">Permissions Matrix</Link> guide.
+
+## Setting granular access controls
+
+When you add a member to your workspace, RudderStack lets you assign any of the three default global roles - <Link to="#read-only">Real-Only</Link>, <Link to="#read-write">Real-Write</Link>, and <Link to="#admin">Admin</Link>. 
+
+Although these permissions provide basic controls, they can end up being too broad or too narrow for certain use-cases. For example, admins cannot restrict access to modifying a certain destination's settings without removing edit permissions for the user entirely. 
+
+With RudderStack's latest granular access control features, admins can lock down business-critical objects to a select list of people. You can also restrict certain users from accessing RudderStack features where the PII (Personally Identifiable Information) is exposed.
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:124]: `You can also restrict certain users from accessing` -> Should we change this to "You can restrict PII access to certain users"? You can only add users to the allowed list and not restrict

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+### Restricting edit permissions for individual objects
+
+This tab lets you specify the list of members having edit permissions to a given object. These permissions include:
+
+- Connecting / disconnecting the object with other objects.
+- Enabling, disabling, or deleting the object
+- Editing or changing the object-specific configuration
+
+<div class="infoBlock">
+Only users with the Admin role have access to the <strong>Permissions</strong> tab. This tab is visible for every source, destination, transformation, tracking plan, and model present in the workspace.
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:143]: There are no granular permissions for transformations and tracking plan

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+<div class="warningBlock">
+Members with Read-Only permissions cannot be added as they do not have permissions to modify an object, by default.
+</div>
+
+## Limiting access to PII-related features
+
+With this feature, you can safeguard your customers' privacy by controlling who has access to the raw event data containing the PII. You can allow everyone on your team to access the PII, or restrict access to a select list of members.
+
+Anyone with the access can view the customers' PII in the <Link to="/dashboard-guides/live-events/">Live Events</Link> and the error logs under the <strong>Events</strong> tab for your <Link to="/dashboard-guides/sources/#source-details">source</Link> or <Link to="/dashboard-guides/destinations/#destination-details">destination</Link>, as shown:
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:169]: Source Events tab doesn't have any error logs. It is present only on destinations

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+
+    - **Anyone on your team**: All the members in your workspace can view the raw event data containing your customers' PII.
+    - **Only people you select**: With this option, only the people you select can view the raw data.
+3. To allow specific members of your team to edit the object, click **Only people you select**, followed by **Add member**.
+4. Finally, select the team members from the drop-down and click **Add Members**, as shown:
+
+<img src="../../assets/rs-cloud/permissions-management-6.png" alt="Add members option" />
+
+<div class="warningBlock">
+If the PII is restricted globally, even the admins will be restricted from viewing the PII.
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:193]: `If the PII is restricted globally` -> PII settings will always be set globally

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 6]
```diff
@@ -0,0 +1,198 @@
+---
+title: "Permissions Management"
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:2]: Do we want to mention that it is a enterprise only feature?

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+description: Detailed description of the permissions management options in RudderStack.
+---
+
+RudderStack's permissions management feature lets you manage users and the permissions assigned to them in your RudderStack workspace.
```
Comment (shekhar-rudder) [inline docs/dashboard-guides/permissions-management/index.mdx:6]: I think we should also talk about permissions logic for performing "connect" actions -> source to destination, source to tracking plan, transformation to destination, model to rETL source

Comment (shekhar-rudder) [reply docs/dashboard-guides/permissions-management/index.mdx:6]: What I meant here was permissions logic is different for connect actions when compared to resource specific actions (edit/delete). So we should explicitly call that out -> For example, you need both source & destination permission to connect/disconnect them OR You can use a model in you rETL source even if you don't have model permissions

Comment (ameypv-rudder) [reply docs/dashboard-guides/permissions-management/index.mdx:6]: @shekhar-rudder do we have a list of such cases where permissions logic is different?

Comment (shekhar-rudder) [reply docs/dashboard-guides/permissions-management/index.mdx:6]: Any connection/linking/de-linking requires permissions on both the resources. The only exception being the SQL model which can used without permissions

**Review by shekhar-rudder — COMMENTED** [review]
LGTM

File: docs/dashboard-guides/permissions-management/index.mdx [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+<img src="../../assets/rs-cloud/permissions-management-3.png" alt="Add members option" />
+
+<div class="warningBlock">
+Members with Read-Only permissions cannot be added as they do not have permissions to modify a resource, by default.
+</div>
+
+### Limiting access to PII-related features
+
+With this feature, you can safeguard your customers' privacy by controlling who has access to the raw event data containing the PII. You can allow everyone on your team to access the PII or allow access to a select list of members.
```
Comment (ishachopra0401) [inline docs/dashboard-guides/permissions-management/index.mdx:156]:
```suggestion
With this feature, you can safeguard your customers' privacy by controlling who has access to the raw event data containing the PII. You can allow everyone on your team to access the PII or restrict the access to a select list of members.
```

File: docs/dashboard-guides/permissions-management/permissions-matrix.mdx [hunk 9]
```diff
@@ -0,0 +1,127 @@
+---
+title: "Permissions Matrix"
+description: Detailed role-specific permissions for Read-Only, Read-Write, and Admin roles.
```
Comment (ishachopra0401) [inline docs/dashboard-guides/permissions-management/permissions-matrix.mdx:3]:
```suggestion
description: Detailed permissions for Read-Only, Read-Write, and Admin roles.
```

File: docs/dashboard-guides/permissions-management/permissions-matrix.mdx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+| View sources list | Yes | Yes | Yes | 
+| View source details | Yes | Yes | Yes | 
+| View tracking plans connected to source | Yes | Yes | Yes |
+| Add new source | No | Yes | Yes | 
+| Delete an existing source | No | Yes | Yes | 
+| Edit source sync schedule | No | Yes | Yes | 
+| Edit source schema | No | Yes | Yes | 
+| Edit source configuration | No | Yes | Yes | 
+| Edit source credentials | No | Yes | Yes | 
+| Edit resources | No | Yes | Yes |
```
Comment (ishachopra0401) [inline docs/dashboard-guides/permissions-management/permissions-matrix.mdx:30]: what does 'resources' here refer to?

File: docs/dashboard-guides/permissions-management/permissions-matrix.mdx [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+<div class="infoBlock">
+Read-Only users can view the events that might contain sensitive user information like PII. In some cases, they can also view the API keys and other API secrets of the destinations.
+</div>
+
+### Syncs
+
+| Permission  | Read-Only | Read-Write | Admin |
+| :-------| :---------| :----------| :------|
+| View sync and history details | Yes  | Yes | Yes  |
+| Sync Extract and Reverse ETL sources | No  | Yes | Yes  |
```
Comment (ishachopra0401) [inline docs/dashboard-guides/permissions-management/permissions-matrix.mdx:99]: we can may be link extract and reverse ETL here, for more clarity

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**